### DISCUSSION TITLE
Indexing for facet search on sub property list

### DISCUSSION THREAD

**Discussion by seanrockster** [ravendb/ravendb#16649] [Q&A]
Apologies if the title is vague.

So I have a product document with a list of attributes, i.e. the attributes are a List<Attribute> on the Product document. An attribute has a name, e.g Colour, and a list of values, e.g. Red, Green, etc.

What I need to do is have an index where the values of the attributes can be indexed and analyzed but I also need to have a facet search so when for example a user searches for 'red', I get back all the products that have an attribute with a value of 'red' but also brings back the list of attribute names so I can create a filter to drill down by attribute name and value.

I've done something similar with Ravendb in the past but the facets where properties on the document. In this case, they are in an array. 

here is my current (fanout) index which appears to work for searching against an attribute value.

```
Map = products => from product in products
            from attribute in product.Attributes
            select new
            {               
                product.Id,
                product.Categories,
                AttributeName = attribute.Name,
                AttributeValue = attribute.Values.Select(x => CreateField("AttributeValue", x.Value))


            };
```

This query in the Studio appears to work.

`from index 'ProductSearch' where search('AttributeValue','*13.5*')`

Thanks.

**Comment by ayende** [comment]
You can do that with dynamic fields, see: https://ravendb.net/docs/article-page/5.4/csharp/indexes/using-dynamic-fields
